### PR TITLE
Add DataSmoothing Toggle next to Filter Buttons

### PR DESCRIPTION
**Pull request by retiutut**

### CODE REVIEW

**Comment by Andrey1994** [comment]
Sorry, I missed it

**Comment by retiutut** [comment]
Other data modes are broken. Let me fix it.

**Comment by daniellasry** [comment]
> Other data modes are broken. Let me fix it.

Yep, this probably crashes on anything that's not cyton serial

File: OpenBCI_GUI/TopNav.pde [hunk 1]
```diff
@@ -311,6 +324,12 @@ class TopNav {
                 filtNotchButton.setIsActive(true);
                 incrementNotchConfiguration();
             }
+            if (smoothingButton.isMouseHere()) {
```
Comment (daniellasry) [inline OpenBCI_GUI/TopNav.pde:327]: probably needs to be
```
if (smoothingButton.isMouseHere() && currentBoard instanceof SmoothingCapableBoard) {
```

**Comment by Andrey1994** [comment]
just run it - smoothing is on by default. lets turn it off

File: OpenBCI_GUI/TopNav.pde [hunk 2]
```diff
@@ -311,6 +332,14 @@ class TopNav {
                 filtNotchButton.setIsActive(true);
                 incrementNotchConfiguration();
             }
+            if (currentBoard instanceof SmoothingCapableBoard) {
+                if (smoothingButton.isMouseHere()) {
+                    smoothingButton.setIsActive(true);
+                    //toggle data smoothing on mousePress for capable boards
+                    ((SmoothingCapableBoard)currentBoard).setSmoothingActive(!((SmoothingCapableBoard)currentBoard).getSmoothingActive());
```
Comment (daniellasry) [inline OpenBCI_GUI/TopNav.pde:339]: Split this up for readability:

```
SmoothingCapableBoard smoothBoard = (SmoothingCapableBoard)currentBoard;
smoothBoard.setSmoothingActive(!smoothBoard .getSmoothingActive());
```

**Pull request merged**